### PR TITLE
[amberscript] Add DRAW_ARRAYS command

### LINKED ISSUE #343: Parse RUN DRAW_ARRAYS commands

### PR DESCRIPTION
**Pull request by dj2**
This CL fills out the draw array parsing for AmberScript.

Fixes #343.

### CODE REVIEW

File: src/amberscript/parser.cc [hunk 1]
```diff
@@ -986,8 +992,50 @@ Result Parser::ParseRun() {
     if (!pipeline->IsGraphics())
       return Result("RUN command requires graphics pipeline");
 
+    token = tokenizer_->NextToken();
+    if (!token->IsString() || token->AsString() != "AS")
+      return Result("missing AS for RUN command");
```
Comment (dneto0) [inline src/amberscript/parser.cc]: The INDEXED case is yet to come.

File: src/amberscript/parser.cc [hunk 2]
```diff
@@ -986,8 +992,50 @@ Result Parser::ParseRun() {
...
+    token = tokenizer_->NextToken();
+    if (!token->IsInteger()) {
+      return Result("invalid START_IDX value for RUN command: " +
+                    token->ToOriginalString());
+    }
+    if (token->AsInt32() < 0)
+      return Result("START_IDX value must be >= 0 for RUN command");
+    uint32_t start_idx = token->AsUint32();
+
+    token = tokenizer_->NextToken();
```
Comment (dneto0) [inline src/amberscript/parser.cc]: This rejects the second DRAW_ARRAY case in the docs.

`RUN <pipeline_name> DRAW_ARRAY AS <topology> START_IDX <value>`

File: src/amberscript/parser.cc (hunk 1 above)
Comment (dj2) [reply src/amberscript/parser.cc]: https://github.com/google/amber/issues/473

File: src/amberscript/parser.cc [hunk 3]
```diff
@@ -986,8 +992,50 @@ Result Parser::ParseRun() {
...
+      return Result("invalid topology for RUN command: " +
+                    token->ToOriginalString());
+    }
+
+    Topology topo = NameToTopology(token->AsString());
+    if (topo == Topology::kUnknown)
+      return Result("invalid topology for RUN command: " + token->AsString());
+
+    token = tokenizer_->NextToken();
+    if (!token->IsString() || token->AsString() != "START_IDX")
```
Comment (dneto0) [inline src/amberscript/parser.cc]: This rejects the first DRAW_ARRAY case in the docs:

`RUN <pipeline_name> DRAW_ARRAY AS <topology>`

File: src/amberscript/parser.cc (hunk 2 above)
Comment (dj2) [reply src/amberscript/parser.cc]: https://github.com/google/amber/issues/474

File: src/amberscript/parser.cc (hunk 3 above)
Comment (dj2) [reply src/amberscript/parser.cc]: https://github.com/google/amber/issues/475

**Pull request merged**